keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,277 @@
+            **policy["solarize_add"], value_range=(0, 255), seed=seed
+        )
+        self.invert = cv_preprocessing.Solarization(
+            **policy["invert"], value_range=(0, 255), seed=seed
+        )
+
+        self.color = cv_preprocessing.RandomColorDegeneration(
+            **policy["color"], seed=seed
+        )
+        self.contrast = cv_preprocessing.RandomContrast(**policy["contrast"], seed=seed)
+        self.brightness = cv_preprocessing.RandomBrightness(
+            **policy["brightness"], seed=seed
+        )
+        self.shear_x = cv_preprocessing.RandomShear(**policy["shear_x"], seed=seed)
+        self.shear_y = cv_preprocessing.RandomShear(**policy["shear_y"], seed=seed)
+        self.translate_x = cv_preprocessing.RandomTranslation(
+            **policy["translate_x"], seed=seed
+        )
+        self.translate_y = cv_preprocessing.RandomTranslation(
+            **policy["translate_y"], seed=seed
+        )
+        self.cutout = cv_preprocessing.RandomCutout(**policy["cutout"], seed=seed)
+
+        self.augmentation_layers = [
+            self.auto_contrast,
+            self.equalize,
+            self.solarize,
+            self.solarize_add,
+            self.invert,
+            self.color,
+            self.contrast,
+            self.brightness,
+            self.shear_x,
+            self.shear_y,
+            self.translate_x,
+            self.translate_y,
+            self.cutout,
+        ]
+        self.auto_vectorize = False
+
+    def _augment(self, sample):
+        sample["images"] = preprocessing_utils.transform_value_range(
+            sample["images"], self.value_range, (0, 255)
+        )
+        augmented_sample = sample
+        for _ in range(self.num_layers):
+            selected_op = tf.random.uniform(
+                (), maxval=len(self.augmentation_layers) + 1, dtype=tf.int32
+            )
+            branch_fns = []
+            for (i, layer) in enumerate(self.augmentation_layers):
+                branch_fns.append((i, lambda: layer(sample)))
+
+            augmented_sample = tf.switch_case(
+                branch_index=selected_op, branch_fns=branch_fns, default=lambda: sample
+            )
+            if self.probability_to_apply is not None:
+                augmented_sample = tf.cond(
+                    tf.random.uniform(shape=(), dtype=tf.float32)
+                    < self.probability_to_apply,
+                    lambda: augmented_sample,
+                    lambda: sample,
+                )
+
+        augmented_sample["images"] = preprocessing_utils.transform_value_range(
+            augmented_sample["images"], (0, 255), self.value_range
+        )
+        return augmented_sample
+
+
+def auto_contrast_policy(magnitude, magnitude_std):
+    return {}
+
+
+def equalize_policy(magnitude, magnitude_std):
+    return {}
+
+
+def solarize_policy(magnitude, magnitude_std):
+    threshold_factor = core.NormalFactorSampler(
+        mean=(magnitude / 10) * 256,
```

Comment:
Should this be 255?

Reply:
Updated - the implementation I based this on used 256 so this was an artifact.  Thanks